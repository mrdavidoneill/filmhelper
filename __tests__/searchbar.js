import "@testing-library/jest-dom";

import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import SearchAppBar from "../components/searchbar";

describe("SearchAppBar component", () => {
  it("renders correctly", () => {
    const onChange = jest.fn();
    const onKeyDown = jest.fn();
    const value = "";
    const { container } = render(
      <SearchAppBar value={value} onChange={onChange} onKeyDown={onKeyDown} />
    );

    expect(container).toMatchSnapshot();
  });

  it("renders the input value correctly", () => {
    const value = "test";
    const { getByLabelText } = render(
      <SearchAppBar value={value} onChange={() => {}} onKeyDown={() => {}} />
    );
    expect(getByLabelText("search").value).toBe(value);
  });

  it("fires onChange event when the input value changes", () => {
    const onChange = jest.fn();
    const { getByLabelText } = render(
      <SearchAppBar value="" onChange={onChange} onKeyDown={() => {}} />
    );
    fireEvent.change(getByLabelText("search"), { target: { value: "test" } });
    expect(onChange).toHaveBeenCalled();
  });

  it("fires onKeyDown event when a key is pressed", () => {
    const onKeyDown = jest.fn();
    const { getByLabelText } = render(
      <SearchAppBar value="" onChange={() => {}} onKeyDown={onKeyDown} />
    );
    fireEvent.keyDown(getByLabelText("search"), { key: "Enter", code: 13 });
    expect(onKeyDown).toHaveBeenCalled();
  });
});
