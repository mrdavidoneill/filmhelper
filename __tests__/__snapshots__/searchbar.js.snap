// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAppBar component renders correctly 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic animate-goright css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-dudo9s-MuiTypography-root"
      >
        Film buddies
      </div>
      <div
        class="css-1orzy9s"
      >
        <div
          class="css-1iz4iln"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary css-7mt4h1-MuiInputBase-root"
        >
          <input
            aria-label="search"
            class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
            placeholder="Searchâ€¦"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
